app/controllers/api/v1/merchants/revenue_controller.rb

```diff
@@ -8,6 +8,20 @@ def date
     render json: {"total_revenue": format_total(total_revenue)}
   end
 
+  def total_revenue_per_merchant
+    if params[:date]
+      total_revenue_per_date
+    else
+      total = Merchant.find(params[:merchant_id]).total_revenue
+      render json: {"revenue": format_total(total)}
+    end
```

Comment:
This is a great workaround but I think it might be better to refactor this if-else to the model level by passing in params and having it check for date there.